src/som/compiler/Parser.java

```diff
@@ -544,7 +545,22 @@ private ExpressionNode evaluation(final MethodGenerationContext mgenc) throws Pa
     ExpressionNode exp = primary(mgenc);
     if (isIdentifier(sym) || sym == Keyword || sym == OperatorSequence
         || symIn(binaryOpSyms)) {
-      exp = messages(mgenc, exp);
+
+      ExpressionNode receiver = exp;
+      SourceCoordinate coord = getCoordinate();
+
+      exp = messages(mgenc, receiver);
+      if (SemiColon == sym) {
+        // Method Cascade
+        List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+        expressions.add(exp);
+
+        while (accept(SemiColon)) {
+          expressions.add(messages(mgenc, receiver));
```

Comment:
This is not correct, I believe.

The receiver of a cascade should be evaluated only once. Otherwise, side effects would be applied multiple times. You could test this with the receiver being a method that increments a counter.

My intuition tells me, that you'll need a new node to implement cascades.

Also, with Truffle, you can't use AST nodes more than once. Each node has a parent pointer, so, that means, no node can have multiple parents.

But, good first step :)
Would be great, if we could integrate that also into TruffleSOM eventually.